impl/pkg/server/util.go

```diff
@@ -2,51 +2,13 @@ package server
 
 import (
 	"net/http"
-	"reflect"
-	"strings"
 
 	"github.com/gin-contrib/cors"
 	"github.com/gin-gonic/gin"
-	"github.com/go-playground/validator/v10"
-	"github.com/goccy/go-json"
 	"github.com/pkg/errors"
 	"github.com/sirupsen/logrus"
 )
 
-var (
-	validate *validator.Validate
-)
-
-func init() {
-	// Instantiate validator.
-	validate = validator.New()
-
-	// Use JSON tag names for errors instead of Go struct field names
-	validate.RegisterTagNameFunc(func(fld reflect.StructField) string {
-		name := strings.SplitN(fld.Tag.Get("json"), ",", 2)[0]
-		if name == "-" {
-			return ""
-		}
-
-		return name
-	})
-}
-
-func Decode(r *http.Request, val any) error {
```

Comment:
was this unused? it's useful for validating http messages into their json structures (with required tags)

Reply:
Seems like it. I'm fine with keeping it if that's your preference.

Reply:
no problem to remove it, can add back later if necessary